src/dashboard/src/components/administration/forms.py

```diff
@@ -432,49 +243,81 @@ class ProcessingConfigurationForm(forms.Form):
-            if 'label' in field:
-                opts['label'] = field['label']
+            opts['label'] = field['label']
+
             if ftype == 'days':
-                if 'min_value' in field:
-                    opts['min_value'] = field['min_value']
+                opts['min_value'] = 0
                 self.fields[choice_uuid] = forms.IntegerField(**opts)
-                if 'placeholder' in field:
-                    self.fields[choice_uuid].widget.attrs['placeholder'] = field['placeholder']
-                self.fields[choice_uuid].widget.attrs['class'] = 'form-control'
-            else:
-                choices = opts['choices'] = list(self.EMPTY_CHOICES)
-                if ftype == 'boolean':
-                    if 'yes_option' in field:
-                        choices.append((field['yes_option'], _('Yes')))
-                    if 'no_option' in field:
-                        choices.append((field['no_option'], _('No')))
-                elif ftype == 'chain_choice':
-                    chain_choices = models.MicroServiceChainChoice.objects.filter(choiceavailableatlink_id=choice_uuid)
-                    ignored_choices = field.get('ignored_choices', [])
-                    for item in chain_choices:
-                        chain = item.chainavailable
-                        if ((chain.description in ignored_choices) or
-                                (not choice_is_available(item, settings))):
-                            continue
-                        choices.append((chain.pk, chain.description))
-                elif ftype == 'replace_dict':
-                    replace_dicts = models.MicroServiceChoiceReplacementDic.objects.filter(choiceavailableatlink_id=choice_uuid)
-                    for item in replace_dicts:
-                        choices.append((item.pk, item.description))
-                elif ftype == 'storage_service':
-                    choices.append(('/api/v2/location/default/{}/'.format(field['purpose']), _('Default location')))
-                    for loc in get_storage_locations(purpose=field['purpose']):
-                        choices.append((loc['resource_uri'], loc['description']))
-                self.fields[choice_uuid] = forms.ChoiceField(widget=Select(attrs={'class': 'form-control'}),
-                                                             **opts)
+                widget = self.fields[choice_uuid].widget
+                widget.attrs['placeholder'] = _('days')
+                widget.attrs['class'] = 'form-control'
+                continue
+
+            choices = opts['choices'] = list(self.EMPTY_CHOICES)
+            if ftype == 'boolean':
+                if 'yes_option' in field:
+                    choices.append((field['yes_option'], _('Yes')))
+                if 'no_option' in field:
+                    choices.append((field['no_option'], _('No')))
+            elif ftype in ('chain_choice', 'replace_dict'):
+                choices.extend(field['options'])
```

Comment:
`field['options']` is now all computed in the server so it gets much simpler here.

Reply:
:clap: